actions.go

```diff
@@ -3,10 +3,258 @@ package neo4j
 import (
 	"fmt"
 
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
 	"github.com/cloudprivacylabs/opencypher/graph"
+	"github.com/fatih/structs"
 	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
 )
 
+type neo4jAction interface {
```

Comment:
It would be cleaner if you split this into two: neo4jAction with Run, and neo4jQueue with Queue.